Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -14,7 +14,7 @@
 
 import struct Foundation.Data
 import XCTest
-@testable import NIO
+@testable import NIOCore
```

Comment:
Should this file (and others) move to a `NIOCoreTests`?

Reply:
Some can move easily (like this one), but plenty cannot because they have awkward cross-dependencies. My plan for now is to ignore the test files and let them blur together, and then to break them apart later on.